Infrastructure/ARM/Deploy.json

```diff
@@ -0,0 +1,259 @@
+                                "roleDefinitionId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions',variables('_roles')[parameters('roles')[div(copyIndex(), variables('lenPrincipalIds'))]])]"
+                            }
+                        }
+                    ]
+
+                },
+                "parameters": {
+                    "principalId": {
+                        "value": [
+                            "[reference(variables('identityId')).principalId]"
+                        ]
+                    },
+                    "roles": {
+                        "value": [
+                            "Key Vault Secrets User",
+                            "Storage Account Queue Contributor",
+                            "Storage Account Blob Data Owner",
+                            "Storage Account Contributor",
+                            "Storage Account Table Contributor"
+                        ]
+                    }
+                }
+            }
+        },
+        {
+            "name": "[variables('storageAccountName')]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2022-05-01",
+            "location": "[variables('location')]",
+            "sku": { "name": "Standard_LRS" },
+            "kind": "StorageV2",
+            "properties": {
+                "isHnsEnabled": true,
+                "isLocalUserEnabled": true,
+                "isSftpEnabled": true
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts/fileServices/shares",
+            "apiVersion": "2022-09-01",
+            "name": "[concat(variables('storageAccountName'), '/default/', variables('shortName'), 'sha')]",
+             "dependsOn": [
+                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+            ]
+            },
+        {
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2020-02-02",
+            "name": "[variables('applicationinsightsname')]",
+            "location": "[variables('location')]",
+            "kind": "web",
+            "properties": {
+                "Application_Type": "web"
+            }
+        },
+        {
+            "type": "Microsoft.Insights/components/pricingPlans",
+            "name": "[concat(variables('applicationinsightsname'),'/', 'Basic')]",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2017-10-01",
+            "dependsOn": [
+                "[resourceId('microsoft.insights/components', variables('applicationinsightsname'))]"
+            ],
+            "properties": {
+                "cap": 1,
+                "planType": "Basic",
+                "warningThreshold": 90
+            }
+        },
+      {
+        "name": "[concat(parameters('company'), '-', variables('projectName'), '-', 'ASP')]",
```

Comment:
company means nothing for an external user. this is just a unique prefix to differentiate between deployments. please rename